lib/pltr/v2/actions/__tests__/ui.test.js

```diff
@@ -224,3 +253,153 @@ describe('cardDialog', () => {
+        const presentState = store.getState().present
+        const allBooks = allBooksSelector(presentState)
+        const totalBooks = allBookIdsSelector(presentState).length
+
+        it('should save the book with its attributes', () => {
+          expect(totalBooks).toBeGreaterThan(initialTotalNumberOfBooks)
+
+          const book = mapValues(allBooks, (value, key) => {
+            return value.title == exampleBookAttributes.title &&
+              value.premise == exampleBookAttributes.premise &&
+              value.genre == exampleBookAttributes.genre &&
+              value.theme == exampleBookAttributes.theme
+              ? true
+              : undefined
+          })
+          expect(Object.keys(book).length).toBeGreaterThan(0)
+        })
+      })
+
+      describe('given setBookDialogClose is dispatched', () => {
+        store.dispatch(setBookDialogClose())
+        const presentState = store.getState().present
+        it('should be back to its initialState', () => {
+          expect(presentState.ui.bookDialog).toMatchObject(initialBookDialogState)
+        })
+      })
+    })
+
+    describe('given setBookDialogOpen is dispatched with bookId (editBook)', () => {
+      const bookId = 1
+      const bookNumber = 1
+      store.dispatch(setBookDialogOpen(bookId, bookNumber))
+      const previousState = store.getState().present
+      const currentBook = allBooksSelector(previousState)[bookId]
+      const initialTotalNumberOfBooks = allBookIdsSelector(previousState).length
+
+      const bookDialogBookId = bookDialogBookIdSelector(previousState)
+      const bookDialogBookNumber = bookDialogBookNumberSelector(previousState)
+      const isOpen = isBookDialogVisibleSelector(previousState)
+
+      it('should open the bookDialog', () => {
+        expect(isOpen).toBeTruthy()
+      })
+
+      it('should have a bookId', () => {
+        expect(bookDialogBookId).not.toBeNull()
+      })
+
+      it('should have a bookNumber', () => {
+        expect(bookDialogBookNumber).not.toBeNull()
+      })
+
+      describe('given editBook is dispatched with id and attributes', () => {
+        const newTitle = 'newBook title'
+        store.dispatch(editBook(currentBook.id, { ...currentBook, title: newTitle }))
+        const presentState = store.getState().present
+        const allBooks = allBooksSelector(presentState)
+        const totalBooks = allBookIdsSelector(presentState).length
+
+        it('should edit the book with the new values', () => {
+          expect(totalBooks).toEqual(initialTotalNumberOfBooks)
+
+          const book = mapValues(allBooks, (value, key) => {
+            return value.title == newTitle &&
+              value.premise == currentBook.premise &&
+              value.genre == currentBook.genre &&
+              value.theme == currentBook.theme
+              ? true
+              : undefined
+          })
+          expect(Object.keys(book).length).toBeGreaterThan(0)
```

Comment:
Again, I'm not sure that this test is good enough.  Perhaps break it into 4 assertions, one for each of the properties you're checking.